workitem/link/link_repository.go

```diff
@@ -145,7 +209,12 @@ func (r *GormWorkItemLinkRepository) Create(ctx context.Context, sourceID, targe
 	}
 
 	if err := r.ValidateTopology(ctx, nil, targetID, *linkType); err != nil {
-		return nil, errs.WithStack(err)
+		return nil, errs.Wrapf(err, "failed to create work item due to invalid topology")
+	}
+	if linkType.Topology == TopologyTree || linkType.Topology == TopologyDependency {
+		if err := r.DetectCycle(ctx, sourceID, targetID, linkTypeID); err != nil {
+			return nil, errs.Wrapf(err, "failed to create work item due to cycle-detection")
```

Comment:
We need to ensure this translates to the correct HTTP error response through a test case in the controller level. I would suggest to use `422 Unprocessable Entity` as the HTTP error response. Ref. https://httpstatuses.com/422

Reply:
@baijum sure, lets care about this when we create the controller logic for this this, shall we?